packages/terra-table/src/subcomponents/HeaderCheckMarkCell.jsx

```diff
@@ -0,0 +1,159 @@
+  /**
+   * @private Callback function not intended for use with this API, but if set pass it through to the element regardless.
+   */
+  onBlur: PropTypes.func,
+  /**
+   * @private Callback function not intended for use with this API, but if set pass it through to the element regardless.
+   */
+  onClick: PropTypes.func,
+  /**
+   * @private Callback function not intended for use with this API, but if set pass it through to the element regardless.
+   */
+  onKeyDown: PropTypes.func,
+  /**
+   * @private Callback function not intended for use with this API, but if set pass it through to the element regardless.
+   */
+  onMouseDown: PropTypes.func,
+};
+
+const defaultProps = {
+  isHidden: false,
+  isDisabled: false,
+  isIndeterminate: false,
+  isSelected: false,
+  isSelectable: false,
+};
+
+const HeaderCheckMarkCell = ({
+  label,
+  alignmentPadding,
+  isHidden,
+  isDisabled,
+  isIndeterminate,
+  isSelected,
+  isSelectable,
+  metaData,
+  onBlur,
+  onClick,
+  onKeyDown,
+  onMouseDown,
+  onSelect,
+  refCallback,
+  ...customProps
+}) => {
+  const attrSpread = { role: 'none' };
+  const attrCheck = {};
+  if (!isHidden && isSelectable) {
+    if (isDisabled) {
+      attrCheck['aria-disabled'] = true;
+    } else {
+      attrSpread.onClick = wrappedOnClickForItem(onClick, onSelect, metaData);
+      attrSpread.onKeyDown = wrappedOnKeyDownForItem(onKeyDown, onSelect, metaData);
+      attrSpread.tabIndex = '0';
+      attrSpread['data-cell-show-focus'] = 'true';
+      attrSpread.onBlur = wrappedEventCallback(onBlur, event => event.currentTarget.setAttribute('data-cell-show-focus', 'true'));
+      attrSpread.onMouseDown = wrappedEventCallback(onMouseDown, event => event.currentTarget.setAttribute('data-cell-show-focus', 'false'));
+    }
+
+    if (isSelected) {
+      attrCheck['aria-checked'] = isIndeterminate ? 'mixed' : true;
+    } else {
+      attrCheck['aria-checked'] = false;
+    }
+  }
+
+  let attrPadding;
+  if (alignmentPadding) {
+    attrPadding = { style: { paddingBottom: alignmentPadding } };
+  }
+
+  const headerCheckMarkCellClasses = cx(
+    'header-cell',
+    { 'is-interactable': !isDisabled && isSelectable },
+  );
+
+  return (
+    <div
+      {...customProps}
+      {...attrSpread}
+      className={customProps.className ? `${headerCheckMarkCellClasses} ${customProps.className}` : headerCheckMarkCellClasses}
+      ref={refCallback}
+      role="columnheader"
+    >
+      <VisuallyHiddenText aria-checked={isSelected} role="checkbox" {...attrCheck} text={label} />
+      <div {...attrPadding} className={cx({ container: !isHidden })}>
+        <div
+          className={`${cx(
+            'checkmark',
+            { 'is-selected': isSelectable && isSelected },
+            { 'is-intermediate': isSelectable && isIndeterminate },
+            { 'is-disabled': isSelectable && isDisabled },
+            { 'is-hidden': isHidden },
+          )} ${customProps.className}`}
```

Comment:
If there aren't any custom props, we'll end up with a trailing space on the className value. It shouldn't hurt anything, but we could trim it?

Reply:
Or ternary it like you're doing elsewhere

Reply:
Updated.